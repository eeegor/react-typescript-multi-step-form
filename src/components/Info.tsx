/**
 * Component Info
 */

import * as React from 'react';
import * as classnames from 'classnames';
import './Info.scss';

export interface Props {
	className?: string;
	label: string | JSX.Element | undefined;
}

export class Info extends React.Component<Props> {
	render(): JSX.Element | null {
		const { className, label } = this.props;

		if (!label || label === '') {
			return null;
		}

		return <small className={classnames('info', className)}>{label}</small>;
	}
}
