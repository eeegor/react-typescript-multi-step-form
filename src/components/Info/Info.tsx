/**
 * Component Info
 */

import React from 'react';
import classnames from 'classnames';
import './Info.scss';

export interface Props {
	className?: string;
	label: string | JSX.Element | undefined;
}

export const Info = (props: Props): JSX.Element | null => {
	const { className, label } = props;

	if (!label || label === '') {
		return null;
	}

	return <small className={classnames('info', className)}>{label}</small>;
};
