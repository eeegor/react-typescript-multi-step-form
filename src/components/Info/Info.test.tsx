import React from 'react';
import renderer from 'react-test-renderer';
import { Info } from './Info';
import { mount } from 'enzyme';

describe('Info', () => {
	it('renders markup correctly', () => {
		const wrapper = <Info label="hello" />;
		const tree = renderer.create(wrapper).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should not output a label with empty value', () => {
		const wrapper = mount(<Info label="" />);
		expect(wrapper.find('.label')).toHaveLength(0);
	});
});
