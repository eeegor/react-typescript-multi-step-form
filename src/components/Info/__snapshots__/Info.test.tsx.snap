// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info renders markup correctly 1`] = `
<small
  className="info"
>
  hello
</small>
`;
